homeassistant/components/husqvarna_automower/calendar.py

```diff
@@ -0,0 +1,114 @@
+"""Creates a calendar entity for the mower."""
+
+from datetime import datetime
+import logging
+
+from aioautomower.model import AutomowerCalendarEvent
+from ical.calendar import Calendar
+from ical.event import Event
+from ical.types.recur import Recur
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import AutomowerConfigEntry
+from .coordinator import AutomowerDataUpdateCoordinator
+from .entity import AutomowerBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: AutomowerConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up lawn mower platform."""
+    coordinator = entry.runtime_data
+    async_add_entities(
+        AutomowerCalendarEntity(mower_id, coordinator) for mower_id in coordinator.data
+    )
+
+
+class AutomowerCalendarEntity(AutomowerBaseEntity, CalendarEntity):
+    """Representation of the Automower Calendar element."""
+
+    _attr_name: str | None = None
+
+    def __init__(
+        self,
+        mower_id: str,
+        coordinator: AutomowerDataUpdateCoordinator,
+    ) -> None:
+        """Set up AutomowerCalendarEntity."""
+        super().__init__(mower_id, coordinator)
+        self._attr_unique_id = mower_id
+        self.calendar = Calendar()
```

Comment:
The `async_get_events` and `event` property are both read calls that should not really have side effects on changing the entity internal members.

You either need to:
- not store `Calendar` as a class member and compute it entirely from `self.mower_attributes.calendar.event` on the fly
- Create the `Calendar` in `_handle_coordinator_update` when the data update coordinator runs

The problem with the current way this works is that `_automower_to_ical_event` has side effects and will mutate the calendar every time it is called which is surprising. It requires the caller to re-initialize the `Calendar` every time, otherwise the list of events on the calendar will grow each time.  My worry is that when the `event` property is called it will keep adding the same events over and over to the calendar.